Basic responsive grid utility, first crack.

BillyH80 opened the pull request:
Find I'm needing to use responsive grids more and more, so taking
a crack at it. Sorry if this is all wrongness or I'm stomping on things already
made or decided. If so let's have a discussion over some warm cookies and
milk and knock it out.

matthewlyle commented:
Hey I never saw this, sorry. Thanks for attempting this. Two issues:

1) We can't make the classes rely on another class like that because they need to be more flexible and able to stack. E.g "on small screens, this is 1 column, on medium its 4 and on large its 2" or something. 
2) We shouldn't be using numbers in our classes like that, it should be tied to a variabile. small/medium/large etc.

See https://github.com/tachyons-css/tachyons/blob/master/src/_widths.css for a good example. (not small seems kind of maybe cool too)

We also need to make sure that whatever we choose as our responsive modifier is flexible to be used for other things. I was thinking __s? e.g .g-1_2__s, .g-1_2__m, .g-1_2__l. But I'm pretty open for anything as long as it's clear that it's a responsive modifier and it doesn't overlap with anything else.

BillyH80 commented:
![screenshot 2017-07-18 11 11 10](https://user-images.githubusercontent.com/16493866/28332573-2a76bd2a-6baa-11e7-8ab0-0810265516f7.png)
![screenshot 2017-07-18 11 11 18](https://user-images.githubusercontent.com/16493866/28332576-2eecfd24-6baa-11e7-8ab9-968a5df7a815.png)
![screenshot 2017-07-18 11 11 28](https://user-images.githubusercontent.com/16493866/28332580-312494da-6baa-11e7-9da9-b318005ddfe8.png)
![screenshot 2017-07-18 11 11 38](https://user-images.githubusercontent.com/16493866/28332583-3316a24c-6baa-11e7-85b8-f2f5a4457fb7.png)

Updated based on feedback.

Not feeling super great about that gigantic assful of classes being set to `width: 100%`, then the subsequent media queries, but not sure how else to best approach this.

Looking at tachyons, their responsiveness appears to be entirely dependent on their width utility, but given that we're using a 3rd party grid system and also `!important` on our width utilities, I don't  think we can replicate how they do it.

Anyway, let me know your thoughts on this approach. Also, I added a `$small-breakpoint` at 640px since that seems to be the one that we use the most.
